scheduler/planner.go

```diff
@@ -243,10 +239,10 @@ func (unit *Unit) RankValue() int64 {
 	} else {
 		// for mainline builds that are more recent, give them a bit
 		// of a bump, to avoid running older builds first.
-		avgLifeTime := lifeTime / time.Duration(length)
+		avgLifeTime := timeInQueue / time.Duration(length)
 
-		if avgLifeTime < time.Duration(48)*time.Hour {
-			unit.cachedValue += priority * unit.distro.GetTimeInQueueFactor() * int64((48*time.Hour - avgLifeTime).Hours())
+		if avgLifeTime < time.Duration(7*24)*time.Hour {
+			unit.cachedValue += priority * int64((7*24*time.Hour - avgLifeTime).Hours())
```

Comment:
really what we want to be doing here is subtracting for each additional hour? maybe?